server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java

```diff
@@ -582,6 +582,8 @@ public void testIssue2991() throws InterruptedException {
     }
 
     public void testSimpleSorts() throws Exception {
+        final int docs = 50;
```

Comment:
To follow up on test cases: I've increased the number of documents (which in turn impacted the optimization thresholds) and the test is now exercising the new flow  much more often (running the same test on `main` fails for me consistently)